piplmesh/account/fields.py

```
@@ -1,16 +1,41 @@
+import datetime
+
 from django.conf import settings
-from django.db.models import fields
+from django.utils.translation import ugettext_lazy as _
+
+import mongoengine
+
+GENDER_CHOICES = (
+    ('male', _('Male')),
+    ('female', _('Female'))
+)
 
 def get_initial_language(request=None):
     return settings.LANGUAGE_CODE
 
-class LanguageField(fields.CharField):
+class LanguageField(mongoengine.StringField):
     def __init__(self, *args, **kwargs):
         kwargs.setdefault('max_length', 5)
         kwargs.setdefault('choices', settings.LANGUAGES)
         kwargs.setdefault('default', get_initial_language)
         
-        super(fields.CharField, self).__init__(*args, **kwargs)
+        super(LanguageField, self).__init__(*args, **kwargs)
+
+class GenderField(mongoengine.StringField):
+    def __init__(self, *args, **kwargs):
+        kwargs.setdefault('max_length', 6)
+        kwargs.setdefault('choices', GENDER_CHOICES)
+        kwargs.setdefault('default', GENDER_CHOICES[1][0])
+        
+        super(GenderField, self).__init__(*args, **kwargs)
+
+class LimitedDateTimeField(mongoengine.DateTimeField):
+    def validate(self, value):
+        super(LimitedDateTimeField, self).validate(value)
         
-    def get_internal_type(self):
-        return "CharField"
\ No newline at end of file
+        try:
+            futureDate = value > datetime.datetime.today() - datetime.timedelta(settings.FUTUREDATE)
+        except:
```

Never catch all exceptions because this could hide possible programming errors.